### PR TITLE
Add metrics calculation to Lightning Module

### LINKED ISSUE #3611: Add metrics calculation to Lightning Module
We only calculate loss during train/val/test in #3272 .  Sometimes metrics are required to calculate the accuracy for evaluating performance of models.
### Steps

- Pass `metrics` to `LightningModuleFromTorch`
- Calculate accuracy by metrics
- Log loss and metrics

### PR DESCRIPTION
**Pull request by zhentaocc**
resolves #3611 

- [x] Pass `metrics` to `LightningModuleFromTorch`
- [x]  Calculate accuracy by metrics
- [x]  Log loss and metrics
- [x]  Test accuracy calculation
- [x]  Pass Jenkins

### CODE REVIEW

File: python/nano/src/bigdl/nano/pytorch/lightning.py [hunk 1]
```diff
@@ -22,18 +23,21 @@
 
 
 class LightningModuleFromTorch(LightningModule):
-    def __init__(self, model: nn.Module, loss: _Loss, optimizer: Optimizer):
+    def __init__(self, model: nn.Module, loss: _Loss, optimizer: Optimizer,
+                 metrics: List[Callable] = None):
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/lightning.py:27]: I suggest directly using [TorchMetrics](https://torchmetrics.readthedocs.io/en/latest/)

File: python/nano/src/bigdl/nano/pytorch/lightning.py [hunk 2]
```diff
@@ -47,19 +51,32 @@ def training_step(self, batch, batch_idx):
...
         return loss
 
     def validation_step(self, batch, batch_idx):
         x, y = batch
         y_hat = self._forward(batch)
         loss = self.loss(y_hat, y)
-        return loss
+        self.log("val_loss", loss, on_epoch=True, prog_bar=True, logger=True)
+        if self.metrics:
+            acc = {"val_acc_%d" % i: metric(y_hat, y) for i, metric in enumerate(self.metrics)}
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/lightning.py:63]: Not all metrics are called "acc", so I think it might be better to use the metric's name if there is one, or use it's class name.

Comment (zhentaocc) [reply python/nano/src/bigdl/nano/pytorch/lightning.py]: Applied your suggestions.

**Comment by zhentaocc** [comment]
Jenkins passed: http://10.112.231.51:18888/job/ZOO-PR-NanoTests/108/

**Comment by yangw1234** [comment]
LGTM

**Comment by zhentaocc** [comment]
## Before the changes
console outputs:
```shell
Epoch 3: 100%|██████████| 8/8 [00:08<00:00,  1.03s/it, loss=3.2, v_num=2]
Testing: 100%|██████████| 4/4 [00:04<00:00,  1.02s/it]--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{}
--------------------------------------------------------------------------------
Testing: 100%|██████████| 4/4 [00:04<00:00,  1.17s/it]
```
tensorboard logging is empty:
image

## After this PR
console outputs:
```shell
Epoch 3: 100%|██████████| 8/8 [00:08<00:00,  1.07s/it, loss=2.97, v_num=3, val/loss=2.610, val/metrics/F1=0.119, val/metrics/Accuracy=0.119]
Testing: 100%|██████████| 4/4 [00:04<00:00,  1.01s/it]--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test/loss': 2.614994764328003,
 'test/metrics/Accuracy': 0.11800000071525574,
 'test/metrics/F1': 0.11800000071525574}
--------------------------------------------------------------------------------
Testing: 100%|██████████| 4/4 [00:04<00:00,  1.11s/it]
```

we can see tensorboard as below:

image  
image

**Comment by zhentaocc** [comment]
@yangw1234 Ready for merge. Jenkins passed http://10.112.231.51:18888/job/ZOO-PR-NanoTests/116/.

**Review by yangw1234 — APPROVED** [review]
